Source/WebCore/platform/graphics/ImageFrame.cpp

```diff
@@ -85,6 +86,7 @@ unsigned ImageFrame::clearImage()
     m_nativeImage->clearSubimages();
     m_nativeImage = nullptr;
     m_decodingOptions = DecodingOptions();
+    m_requiresSynchronousDecode = true;
```

Comment:
I do not this is needed. And I think it is incorrect as well. We clear the image frame under memory pressure. The `decodingMode` should be specified by the client of the image. We should not assume once the frame is deleted, it has to be synchronously decoded always. 

The logic to reset this flag once `BitmapImage::s_requireSynchronousRedecoding` is reset to false is not reflected in the name and it is not obvious.

Reply:
Without this flag on the ImageFrame, there's no easy way to track whether this is the first paint since the tab came back to the foreground. We set the static s_requireSynchronousRedecode back to false when the tab goes into the foreground, but the first repaint might be some time later, because we've shown the page snapshot initially. So by the time the user interacts with the page to cause an actual first repaint, we don't know that we need to do decode synchronously.